### DISCUSSION TITLE
Memory LEAKS

### DISCUSSION THREAD

**Discussion by Dmariano06** [codam-coding-college/MLX42#140] [Q&A] [answered]
Hi everyone, I’m working on the so_Long project with MLX and I’ve been dealing with these leaks that keep coming up I havn’t been able to track their source and i’m wondering if it’s normal to have so many, as it seems like the leaks might be coming from my drivers. Can I validate my project with this issue?
thank you very much

Direct leak of 65536 byte(s) in 1 object(s) allocated from:
    #0 0x7ff00ee85517 in __interceptor_aligned_alloc ../../../../src/libsanitizer/lsan/lsan_interceptors.cpp:148
    #1 0x7ff001cd61df  (/usr/lib/wsl/drivers/iigd_dch.inf_amd64_91a9902d227a84da/libLLVM-14.so+0x23d1df)

Direct leak of 3012 byte(s) in 1 object(s) allocated from:
    #0 0x7ff00ee85a92 in operator new(unsigned long) ../../../../src/libsanitizer/lsan/lsan_interceptors.cpp:248
    #1 0x7feffddf8cba  (/usr/lib/wsl/drivers/iigd_dch.inf_amd64_91a9902d227a84da/libigc.so+0x8efcba)
    #2 0x7feffddfcc7c  (/usr/lib/wsl/drivers/iigd_dch.inf_amd64_91a9902d227a84da/libigc.so+0x8f3c7c)
    #3 0x7ff001132dc0  (<unknown module>)
    #4 0x7ff00112967f  (<unknown module>)
    #5 0x7ff0011278c0  (<unknown module>)
    #6 0x7ff001122dec  (<unknown module>)
    #7 0x7ff0011cf8da  (<unknown module>)
    #8 0x7ff0011cf5f2  (<unknown module>)
    #9 0x7ff0010f29c8  (<unknown module>)
    #10 0x7ff001e532a1  (/usr/lib/wsl/drivers/iigd_dch.inf_amd64_91a9902d227a84da/libLLVM-14.so+0x3ba2a1)
    #11 0x6140000097cf  (<unknown module>)
SUMMARY: LeakSanitizer: 107080 byte(s) leaked in 35 allocation(s).

**Comment by W2Wizard** [comment, marked as answer]
> Can I validate my project with this issue?

Yes eitherway, even if MLX42 were to have leaks or some drivers fucks up. It's only a problem if your own code causes the leak or misuses MLX functions that then lead to a direct leak.

If you run it on a school computer and get no leaks you should be 101% sure its all fine.

**Reply by Dmariano06** [reply]
thank you very much for your quick response , may I share the main part of my code with you so that you can tell me if i’m initializing my MLX incorrectly or using the images wrong?

**Reply by W2Wizard** [reply]
Go for it 👍
